jps-plugin/src/org/intellij/erlang/jps/builder/ErlangPrepareDependenciesBuilder.java

```diff
@@ -0,0 +1,270 @@
+ * limitations under the License.
+ */
+
+package org.intellij.erlang.jps.builder;
+
+import com.intellij.openapi.util.JDOMUtil;
+import com.intellij.util.Function;
+import com.intellij.util.SystemProperties;
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.util.graph.GraphGenerator;
+import com.intellij.util.xmlb.SkipDefaultValuesSerializationFilters;
+import com.intellij.util.xmlb.XmlSerializationException;
+import com.intellij.util.xmlb.XmlSerializer;
+import org.jdom.Document;
+import org.jdom.JDOMException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.jps.builders.BuildOutputConsumer;
+import org.jetbrains.jps.builders.DirtyFilesHolder;
+import org.jetbrains.jps.incremental.CompileContext;
+import org.jetbrains.jps.incremental.ProjectBuildException;
+import org.jetbrains.jps.incremental.TargetBuilder;
+import org.jetbrains.jps.incremental.messages.BuildMessage;
+import org.jetbrains.jps.incremental.messages.CompilerMessage;
+import org.jetbrains.jps.model.JpsProject;
+import org.jetbrains.jps.model.java.JavaSourceRootProperties;
+import org.jetbrains.jps.model.java.JavaSourceRootType;
+import org.jetbrains.jps.model.module.JpsModule;
+import org.jetbrains.jps.model.module.JpsTypedModuleSourceRoot;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+public class ErlangPrepareDependenciesBuilder extends TargetBuilder<ErlangSourceRootDescriptor, ErlangPrepareDependenciesTarget> {
+  public static final String DEPENDENCIES_TREE_FILE_PATH = "erlang-builder/deps-tree.xml";
+  private static final String NAME = "Prepare dependence";
+
+  public ErlangPrepareDependenciesBuilder() {
+    super(Collections.singletonList(ErlangPrepareDependenciesTargetType.INSTANCE));
+  }
+
+  @Override
+  public void build(@NotNull ErlangPrepareDependenciesTarget target,
+                    @NotNull DirtyFilesHolder<ErlangSourceRootDescriptor, ErlangPrepareDependenciesTarget> holder,
+                    @NotNull BuildOutputConsumer outputConsumer,
+                    @NotNull CompileContext context) throws ProjectBuildException, IOException {
+    ErlangDependencyTree dependencyTree = loadDependencyTree(context);
+    if (dependencyTree == null) {
+      addPrepareDependenciesFailedMessage(context);
+      return;
+    }
+
+    List<String> dirtyErlangFilePaths = new DirtyFileProcessor<String, ErlangPrepareDependenciesTarget>() {
+      @Nullable
+      @Override
+      protected String getDirtyElement(@NotNull ErlangPrepareDependenciesTarget target,
+                                       @NotNull File file,
+                                       @NotNull ErlangSourceRootDescriptor root) throws IOException {
+        return isSourceOrHeader(file) ? file.getAbsolutePath() : null;
+      }
+    }.collectDirtyElements(holder);
+    prepareDependencies(target.getProject(), context, dependencyTree, dirtyErlangFilePaths);
```

Comment:
I don't see any benefit in extracting this ```prepareDependencies``` method.

Think about what your ```org.intellij.erlang.jps.builder.ErlangPrepareDependenciesBuilder``` is meant to do.

First, it should compute all dirty files, second, it should write these file paths to an XML. It's that simple. Why not state it in code? 

Instead, you have these two steps' logic mixed in not one, but two methods. You load your dependency tree, compute initial dirty files set, pass it all to another method which then computes dirty files and writes them to your xml files - isn't it weird?